src/Newton.jl

```diff
@@ -113,7 +113,7 @@ function newton(Fhandle, Jhandle, x0, options:: NewtonPar{T}; normN = norm) wher
 	@unpack tol, maxIter, verbose, linesearch = options
 
 	# Initialise iterations
-	x = copy(x0)
+	x = convert(typeof(Fhandle(x0)),copy(x0))
```

Comment:
I am not sure about this, you should enforce this in your function. It would break the use of `ApproxFun`.

Reply:
if I don't have this change the unit tests a7e4af1 fail with
```
ERROR: LoadError: Can't differentiate `setindex!`
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] #setindex!#369(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::TrackedArray{…,Array{Float64,1}}, ::Tracker.TrackedReal{Float64}, ::Int64) at /home/gszep/.julia/packages/Tracker/SAr25/src/lib/array.jl:65
 [3] setindex!(::TrackedArray{…,Array{Float64,1}}, ::Tracker.TrackedReal{Float64}, ::Int64) at /home/gszep/.julia/packages/Tracker/SAr25/src/lib/array.jl:65
 [4] minus!(::TrackedArray{…,Array{Float64,1}}, ::Array{Tracker.TrackedReal{Float64},1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.1/LinearAlgebra/src/generic.jl:1174
 [5] #newton#59(::typeof(norm), ::Function, ::getfield(Main, Symbol("##3#4")){getfield(Main, Symbol("##7#8")),Tracker.TrackedReal{Float64}}, ::getfield(PseudoArcLengthContinuation, Symbol("##61#62")){getfield(Main, Symbol("##3#4")){getfield(Main, Symbol("##7#8")),Tracker.TrackedReal{Float64}}}, ::TrackedArray{…,Array{Float64,1}}, ::NewtonPar{Tracker.TrackedReal{Float64},DefaultLS,DefaultEig}) at /home/gszep/.julia/dev/PseudoArcLengthContinuation/src/Newton.jl:137
 [6] newton at /home/gszep/.julia/dev/PseudoArcLengthContinuation/src/Newton.jl:113 [inlined]
 [7] #newton#60 at /home/gszep/.julia/dev/PseudoArcLengthContinuation/src/Newton.jl:155 [inlined]
 [8] newton at /home/gszep/.julia/dev/PseudoArcLengthContinuation/src/Newton.jl:154 [inlined]
 [9] continuation(::getfield(Main, Symbol("##7#8")), ::TrackedArray{…,Array{Float64,1}}, ::Tracker.TrackedReal{Float64}) at /home/gszep/Documents/code/fold-inference/continuation.jl:10
 [10] loss(::Function, ::TrackedArray{…,Array{Float64,1}}, ::Tracker.TrackedReal{Float64}) at /home/gszep/Documents/code/fold-inference/main.jl:13
 [11] top-level scope at none:0
 [12] include at ./boot.jl:326 [inlined]
 [13] include_relative(::Module, ::String) at ./loading.jl:1038
 [14] include(::Module, ::String) at ./sysimg.jl:29
 [15] exec_options(::Base.JLOptions) at ./client.jl:267
 [16] _start() at ./client.jl:436
in expression starting at /home/gszep/Documents/code/fold-inference/main.jl:25
```